airflow-core/src/airflow/models/deadline.py

```diff
@@ -98,6 +98,61 @@ def _determine_resource() -> tuple[str, str]:
             f"{self.deadline_time} or run: {self.callback}({callback_kwargs})"
         )
 
+    @classmethod
+    @provide_session
+    def remove_deadlines(cls, *, session: Session, conditions: dict[Column, Any]) -> int:
+        """
+        Remove deadlines from the table which match the provided conditions and return the number removed.
+
+        NOTE: This should only be used to remove deadlines which are associated with
+            successful dagruns. If the deadline was missed, move it to the `missed_deadlines`
+            table after executing the callback.
```

Comment:
Is the move intended to happen here? I wonder if we should rename this thing to something like `process_deadlines` since it feels like a bit more than deletion might happen here.

Reply:
Yeah synced with Ramit about the table vs column, that all sounds good. If that happens elsewhere I guess this function name is good enough. Although, I will note that it is still quite conditional in some cases not all, or even any, deadlines will be deleted. So it's more like a prune (which is a name you suggested when we chatted in Slack and I like that).

Reply:
Yeah, changed to prune in the last revision, going to resolve this one.  It's a much better name and implies that we may or may not be actually deleting, depending on the condition.